### DISCUSSION TITLE
Typo in globals.h?

### DISCUSSION THREAD

**Discussion by airnocker** [PlummersSoftwareLLC/NightDriverStrip#432] [Q&A] [answered]
Line 236  // LEDs, how many, on how many channels, laid out into how many **_fang_**/rings, and so on.

Do you mean "fans/rings"?  i.e. like how many fans (straight lines) of strips?

airnocker (Alden G)

**Comment by rbergen** [comment]
Yep, that looks like a typo to me. "Fans" should probably be interpreted as "devices that provide cooling to desktop computers", as that is one of the projects Dave put together with this before he open sourced it.

**Reply by robertlipe** [reply]
Almost certainly. 

If you have something you're sure is a typo or needs improved words that you can provide, I'd suggest just clicking edit (the little pencil) and preparing a change. If it passes review (agreement that you left it better than you found it), someone with approval privs (almost always Rutger) can then just click a button to apply it. Doc changes are much easier to integrate than code changes, which currently, to my knowledge, still require a human  (again, usually Rutger) to touch-test them on a reasonable subset of the hardware population before applying them.

It's been my own experience that doc changes are pretty easy to get approved and integrated. As we get more eyes on them, we have both the blessing of more hands to fix them and the curse (well, that's perhaps overstated) of more eyeballs actually reading them and possibly getting mislead by glitches. I think more hands on the doc in this project will help it a lot.

It's a problem in every code base I've worked with that the doc is read by people new to the project and they expect it to make sense. The people that have worked on the code forever, perhaps the same people that introduced changes that invalidate the doc, will develop blindness to the doc because THEY aren't the ones reading it. This is why input from new members is particularly valuable and this project is pretty good about welcoming input.

I still don't really get "rings", but it's pretty clear that "fangs"

**Comment by airnocker** [comment]
@rbergen, oh yeah sure, FANS, @davepl computer fan illuminations YT mentions with WS2812 strips!

Thanks for information on editing docs.  I'll do that.

**Comment by airnocker** [comment]
So I'm only working with a local copy of NightDriverStrip.  I can edit it and save easily.  Don't yet know the ins and outs of being a "collaborative actor".

**Comment by robertlipe** [comment, marked as answer]
Good. For easy changes that don't need compilation/testing, like a few
words in a doc, just clicking the edit button on the page of, say,
https://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/include/globals.h
and then letting it create the branch and pull request for you is probably
fine.

For things that need you to build them and that'll thus need non-trivial
review and testing from Rutger or Dave to be integrated, you'll need to
build it yourself to be sure everything works, then open a pull request
from your local, tested, tree..  Use your favorite search engine for
[github make pull request] and then boil down to a combination of operating
system and tools that you find comfortable.  For example, there is a
'GitHub Desktop' thingy that some people like to help with that process. I
personally use the 'gh <https://cli.github.com/>' tool, which is admittedly
not for everyone. There are professional tools that help once you start to
have professional-level problems. There's GH's own official guidance.
<https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request>
There
are LOTS of pages on this, so I'll just point you in the right direction.

**Reply by airnocker** [reply]
I made the change with the link you provided on dave's nightdriverstrip page, very easy to do, thanks.
I have GitHub Desktop but only because I thought it was GitCMD, now I have both, and will defer from using the Desktop thingy since I've still got a lot of spaghetti on my plate.  I'm also fairly new to GitHub beyond downloading zip files.

**Comment by rbergen** [comment]
@airnocker I've corrected this typo in a branch in my fork with (otherwise unrelated) changes I'm currently working on. As such, the correction will get pulled in with an upcoming larger update to the project.

**Reply by airnocker** [reply]
Cool, glad you found the root cause.  I certainly appreciate your looking into this.

What do I have to do for my local copy of NightDriverStrip to benefit from this change once the project is updated?

**Reply by rbergen** [reply]
That depends a little on how you created your local copy. Could you run a "git remote -v" for me in the `NightDriverStrip` project directory on your development machine, and tell me what it says?

**Reply by airnocker** [reply]
@rbergen, I'll try that later today and post the results.  I assume I have to be logged into git for this to yield any meaningful results.  (ain't ever logged onto git within PIO before, although it always asks if I want to do so and I say 'Cancel').

**Reply by rbergen** [reply]
@airnocker This may sound a bit pedantic which is not my intention, but I can't help how others named things:

- GitHub is a platform that can be used to host source code and collaborate with others on one codebase, and it indeed is the platform NightDriverStrip uses for that purpose.
- git is a "development machine" tool to keep track of source code changes, and potentially synchronize those with platforms like GitHub.

It's the combination that can yield very powerful collaborative development experiences, one gets their head around both - which I will be the first to say involves a pretty steep learning curve.

I'm pretty sure the login PIO asks for concerns GitHub. My request concerns the output of git - which may tell us it's not even installed, but that in itself would also me help answering your original question.

**Reply by airnocker** [reply]
@rbergen I replied with the results of the git command and do understand git vs GitHub.  I may have posted the git results in one of our other threads because I don't see it here.

**Reply by rbergen** [reply]
I'm not sure where your git command output ended up then. I searched my email and didn't receive a notification of a comment mentioning git in the past 48 hours or so - except for this one.

**Reply by airnocker** [reply]
Hmm, it appears I took a screenshot of the 'git remote -v' result and then failed to "send" it to you. My apologies.  So here it is:

![git remote -v](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/31143220/5592bf3d-c39e-4434-98c5-c7e12c7794af)

**Reply by rbergen** [reply]
No worries. In that case a "git pull" should do it in principle. Do note that if you have any uncommitted local modifications to your existing local version, git may insist on you reverting or stashing them. If you have any _committed_ local changes you may have (a) merge conflict(s) to resolve.
